pkg/apis/core/validation/ratcheting.go

```diff
@@ -60,4 +60,31 @@ var RatchetingPodSpecValidations = apimachineryvalidation.RatchetingValidators{
 		}
 		return allErrs
 	},
+
+	// duplicate envvar names
+	func(data interface{}, fldPath *field.Path) field.ErrorList {
+		allErrs := field.ErrorList{}
+		spec := data.(*core.PodSpec)
+		for i, c := range spec.InitContainers {
+			names := sets.NewString()
+			for j, e := range c.Env {
```

Comment:
nit: seems like these loops could be a subroutine.

Reply:
cleaned up the loops